src/modules/lorem/index.ts

```diff
@@ -332,7 +358,26 @@ export class LoremModule {
 
     const method = this.faker.helpers.arrayElement(methods);
 
-    return `${this[method]()}`;
+    if (typeof options === 'number') {
+      options = { length: options };
+    }
+
+    if (options.length == null) {
+      return `${this[method]()}`;
+    }
+
+    const length = this.faker.helpers.rangeToNumber(options.length);
+
+    if (length <= 0) {
+      return '';
+    }
+
+    let text = '';
+    do {
+      text = `${text}${this[method]()} `;
+    } while (text.length <= length + 1);
```

Comment:
```suggestion
    } while (text.length < length - 1);
```

If `text.length === length - 1` then the substring below doesn't remove anything and appends the '.' string => `text.length === length`. It would end in `foo .` but that would still be the case even if you append another word to it, because it would get removed via the substring call.

---

Should we deduct the `-1` from `length` early?

````ts
// deduct one from the length for the final dot
length--;
````

---

Alternatively see this comment: https://github.com/faker-js/faker/pull/2227#discussion_r1299768500

Reply:
I implement it similar to the way you described in https://github.com/faker-js/faker/pull/2227#discussion_r1299768500.